service/create/service.go

```diff
@@ -1281,12 +1310,40 @@ func (s *Service) deleteFunc(obj interface{}) {
 		s.logger.Log("error", fmt.Sprintf("could not delete security group '%s': '%#v'", ingressSGInput.GroupName, err))
 	}
 
-	// Delete VPC peering connection.
-	vpcPeeringConection := &awsresources.VPCPeeringConnection{
+	var vpcPeeringConection resources.ResourceWithID
+	vpcPeeringConection = &awsresources.VPCPeeringConnection{
 		VPCId:     vpcID,
 		PeerVPCId: cluster.Spec.AWS.VPC.PeerID,
-		AWSEntity: awsresources.AWSEntity{Clients: clients},
+		AWSEntity: awsresources.AWSEntity{
+			Clients:     clients,
+			HostClients: hostClients,
+		},
+	}
+
+	// Delete Guest VPC Route.
+	for _, privateSubnetSpec := range cluster.Spec.AWS.VPC.PrivateSubnets {
+		privateSubnet := &awsresources.Subnet{
```

Comment:
Also here, `DefaultSubnetConfig()` and `DefaultRouteTableConfig()` might help, i've seen this more often..